pilot/cmd/pilot-agent/main.go

```diff
@@ -305,6 +295,39 @@ var (
 	}
 )
 
+func getPilotSAN(domain string, ns string) []string {
+	var pilotSAN []string
```

Comment:
If we are doing this, my main request is to provide a way to explicitly set the SAN - right now there is a lot of guessing and that's never good for security. In particular I want to have a way to specify a DNS SAN, for cases where pilot is running with standard certificates ( like a managed version, outside of the cluster ).

Reply:
The users can specify the TrustDomain in the command line, which is used to generate the SAN.
If you think that's not explicit - we could allow users to specify the SAN in the command line, then you may need to make sure the consistency between the SAN value and the TrustDomain value used by other components.

BTW, can we add a TODO here for DNS SAN? Supporting this feature for Pilot seems not quite related to this PR.

Reply:
I agree, it was just an idea. Technically for DNS we don't need a flag - it's the domain name, just need to verify it the standard way. 

My only concern remains hardcoding the pilot service account - but we can live with that in 1.1.